### PR TITLE
Island: Add IEventRepository

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2180 

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] ~Was the CHANGELOG.md updated to reflect the changes?~
* [x] ~Was the documentation framework updated to reflect the changes?~
* [x] ~Have you checked that you haven't introduced any duplicate code?~

## Testing Checklist

* [x] ~Added relevant unit tests?~
* [x] ~Have you successfully tested your changes locally? Elaborate:~
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/monkey_island/cc/repository/i_event_repository.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import Sequence
+
+from common.events import AbstractAgentEvent
+
+
+class IEventRepository(ABC):
+    """A repository used to store and retrieve Events objects"""
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/repository/i_event_repository.py:8]:
```suggestion
    """A repository used to store and retrieve event objects"""
```

File: monkey/monkey_island/cc/repository/i_event_repository.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @abstractmethod
+    def save_event(self, event: AbstractAgentEvent):
+        """
+        Save event to the repository
+
+        :param event: The event that we are going to store
+        :raises StorageError: If an error occured while attempting to save an Event
+        """
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/repository/i_event_repository.py:17]:
```suggestion
        """
        Save an event to the repository

        :param event: The event to store in the repository
        :raises StorageError: If an error occured while attempting to save an event
        """
```

File: monkey/monkey_island/cc/repository/i_event_repository.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    @abstractmethod
+    def get_events(self) -> Sequence[AbstractAgentEvent]:
+        """
+        Return all events that are stored in the repository
+
+        :return: All known Events
+        :raises RetrievalError: If an error occured while attempting to retrieve the events
+        """
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/repository/i_event_repository.py:26]:
```suggestion
        """
        Retrieve all events stored in the repository

        :return: All stored events
        :raises RetrievalError: If an error occured while attempting to retrieve the events
        """
```

File: monkey/monkey_island/cc/repository/i_event_repository.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    @abstractmethod
+    def reset(self):
+        """
+        Removes all data from the repository
+
+        :raises RemovalError: If an error occurred while attempting to remove all `Events` from the
+                              repository
+        """
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/repository/i_event_repository.py:35]:
```suggestion
        """
        Remove all data from the repository

        :raises RemovalError: If an error occurred while attempting to remove all events from the
                              repository
        """
```

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Just some minor docstring changes

File: monkey/monkey_island/cc/repository/i_event_repository.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import Sequence
+
+from common.events import AbstractAgentEvent
+
+
+class IEventRepository(ABC):
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/repository/i_event_repository.py:7]: We need the ability to get events by type, tag, or source. Add 3 methods for those.

**Review by cakekoa — APPROVED** [review]

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**